lib/git-metadata-preprocessor/extension.rb

```diff
@@ -0,0 +1,43 @@
+require 'asciidoctor/extensions' unless RUBY_ENGINE == 'opal'
+
+require 'rugged'
+
+class GitMetadataPreprocessor < Asciidoctor::Extensions::Preprocessor
+  def process document, reader
+
+    begin
+      repo = Rugged::Repository.discover(".")
+    rescue
+      $stderr.puts('Failed to find repository, git-metadata extension terminating')
+      exit
```

Comment:
I don't think we want to exit the Ruby program (Asciidoctor) if one extension fails

Reply:
Sounds fair to me.